bookwyrm/models/author.py

```diff
@@ -67,9 +68,28 @@ def get_remote_id(self):
         """editions and works both use "book" instead of model_name"""
         return f"https://{DOMAIN}/author/{self.id}"
 
-    activity_serializer = activitypub.Author
-
     class Meta:
-        """sets up postgres GIN index field"""
+        """sets up indexes and triggers"""
+
+        triggers = [
+            pgtrigger.Trigger(
+                name="reset_search_vector_on_author_edit",
+                when=pgtrigger.After,
+                operation=pgtrigger.UpdateOf("name"),
+                func=format_trigger(
+                    """WITH updated_books AS (
+                         SELECT book_id
+                         FROM bookwyrm_book_authors
+                         WHERE author_id = new.id
```

Comment:
Now that this bug is fixed, I think the migration should re-calculate the search vector for all books because it may have missed some author name changes.

Reply:
First of all, thanks for the review! I addressed most outstanding items (there's still the question about #3274 pending).

Secondly: if I understand correctly, your suggestion here is to include in the migration something like: `UPDATE bookwyrm_book SET search_vector = ''`?

Reply:
Yes, exactly

Reply:
I'll admit I've failed to add it so far because it's a destructive operation that I wanted to double-check. I'll take a stab after reviewing Hugh's pending PR, but help welcome in the meantime, if you feel so inclined! (no obligation either way)

Reply:
Added to the migration and and tested